app/lib/tariff_synchronizer.rb

```diff
@@ -178,33 +178,37 @@ def rollback(rollback_date, keep: false)
         Sequel::Model.db.transaction do
           # Delete actual data
           oplog_based_models.each do |model|
-            model.operation_klass.where { operation_date > date_for_rollback }.delete
+            model.operation_klass.where(Sequel.lit('operation_date > ?', date_for_rollback)).delete
           end
 
           if keep
             # Rollback TARIC
-            TariffSynchronizer::TaricUpdate.applied_or_failed.where { issue_date > date_for_rollback }.each do |taric_update|
```

Comment:
In this case, Sequel used **virtual row blocks** syntax.
For example:

    Book.where{stars >= 3}
That is not efficient, since behind the scenes, Sequel uses method_missing is to instantiate identifiers and function objects.

Most of the time it does not matter,
but in this case, it is inside a loop, inside a loop, so it does matter (I've tested it).

Reply:
Since there are other broken tests, I rolled back this improvement to exclude this as a possible cause.